content/TruenasApps/UsingCustomApp.md

```diff
@@ -48,138 +35,1199 @@ If you are unsure about any configuration settings, review the [Install Custom A
+        required: true
+    deploy:
+      resources:
+        limits:
+          cpus: "2"
+          memory: "4294967296"
+    environment:
+      ALLOW_EMPTY_PASSWORD: "no"
+      REDIS_PASSWORD: password
+      REDIS_PORT_NUMBER: "6379"
+    healthcheck:
+      test:
+        - CMD-SHELL
+        - redis-cli -h 127.0.0.1 -p 6379 -a $$REDIS_PASSWORD ping | grep -q PONG
+      timeout: 5s
+      interval: 10s
+      retries: 30
+      start_period: 10s
+    image: bitnami/redis:7.4.1
+    networks:
+      default: null
+    restart: unless-stopped
+    security_opt:
+      - no-new-privileges
+    user: "1001:0"
+    volumes:
+      - type: volume
+        source: tmp
+        target: /tmp
+        volume: {}
+      - type: volume
+        source: bitnami_redis_data
+        target: /bitnami/redis/data
+        volume: {}
+networks:
+  default:
+    name: ade25cf7c52e909aeef17c9aaee373aa_default
+volumes:
+  bitnami_redis_data:
+    name: ade25cf7c52e909aeef17c9aaee373aa_bitnami_redis_data
+  nextcloud-data:
+    name: ade25cf7c52e909aeef17c9aaee373aa_nextcloud-data
+  nextcloud-html:
+    name: ade25cf7c52e909aeef17c9aaee373aa_nextcloud-html
+  nextcloud-postgres-data:
+    name: ade25cf7c52e909aeef17c9aaee373aa_nextcloud-postgres-data
+  tmp:
+    name: ade25cf7c52e909aeef17c9aaee373aa_tmp
+configs:
+  ix-update-hosts-script.sh:
+    name: ade25cf7c52e909aeef17c9aaee373aa_ix-update-hosts-script.sh
+    content: |
+      #!/bin/bash
+      set -e
+      config_file="/var/www/html/config/config.php"
+
+      echo "Updating database and redis host in config.php"
+      sed -i "s/\('dbhost' => '\)[^']*postgres:5432',/\1postgres:5432',/" "$$config_file"
+      occ config:system:set redis host --value="redis"
+  limitrequestbody.conf:
+    name: ade25cf7c52e909aeef17c9aaee373aa_limitrequestbody.conf
+    content: |2
+      LimitRequestBody 3221225472
+  occ:
+    name: ade25cf7c52e909aeef17c9aaee373aa_occ
+    content: |
+      #!/bin/bash
+      uid="$$(id -u)"
+      gid="$$(id -g)"
+      if [ "$$uid" = '0' ]; then
+        user='www-data'
+        group='www-data'
+      else
+        user="$$uid"
+        group="$$gid"
+      fi
+      run_as() {
+        if [ "$$(id -u)" = 0 ]; then
+          su -p "$$user" -s /bin/bash -c "php /var/www/html/occ $$(printf '%q ' "$$@")"
+        else
+          /bin/bash -c "php /var/www/html/occ $$(printf '%q ' "$$@")"
+        fi
+      }
+      run_as "$$@"
+  opcache.ini:
+    name: ade25cf7c52e909aeef17c9aaee373aa_opcache.ini
+    content: |
+      opcache.memory_consumption=128
+  php.ini:
+    name: ade25cf7c52e909aeef17c9aaee373aa_php.ini
+    content: |
+      max_execution_time=30{{< /codeblock >}}
+
+Alternatively, you can use [`dockerfile`](https://docs.docker.com/reference/compose-file/build/#dockerfile) to set an existing Dockerfile stored on your TrueNAS system.
+
+To specify the absolute path to a Dockerfile, remove this `dockerfile_inline` section of the `build` element:
```

Comment:
same comment about capitalzing dokerfile as it is not a proper noun.
```suggestion
Alternatively, you can use [`dockerfile`](https://docs.docker.com/reference/compose-file/build/#dockerfile) to set an existing dockerfile stored on your TrueNAS system.

To specify the absolute path to a dockerfile, remove this `dockerfile_inline` section of the `build` element:
```

Reply:
same reply